oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/authentication/JwtAuthenticationConverter.java

```diff
@@ -16,57 +16,53 @@
 
 package org.springframework.security.oauth2.server.resource.authentication;
 
-import java.util.Arrays;
 import java.util.Collection;
-import java.util.Collections;
-import java.util.stream.Collectors;
 
 import org.springframework.core.convert.converter.Converter;
 import org.springframework.security.authentication.AbstractAuthenticationToken;
 import org.springframework.security.core.GrantedAuthority;
-import org.springframework.security.core.authority.SimpleGrantedAuthority;
 import org.springframework.security.oauth2.jwt.Jwt;
-import org.springframework.util.StringUtils;
+import org.springframework.util.Assert;
 
 /**
  * @author Rob Winch
  * @author Josh Cummings
  * @since 5.1
  */
 public class JwtAuthenticationConverter implements Converter<Jwt, AbstractAuthenticationToken> {
-	private static final String SCOPE_AUTHORITY_PREFIX = "SCOPE_";
-
-	private static final Collection<String> WELL_KNOWN_SCOPE_ATTRIBUTE_NAMES =
-			Arrays.asList("scope", "scp");
-
+	private Converter<Jwt, Collection<GrantedAuthority>> grantedAuthoritiesConverter
+			= new JwtGrantedAuthoritiesConverter();
 
+	@Override
 	public final AbstractAuthenticationToken convert(Jwt jwt) {
 		Collection<GrantedAuthority> authorities = extractAuthorities(jwt);
 		return new JwtAuthenticationToken(jwt, authorities);
 	}
 
+	/**
+	 * Extracts the {@link GrantedAuthority}s from scope attributes typically found in a {@link Jwt}
+	 *
+	 * @param jwt The token
+	 * @return The collection of {@link GrantedAuthority}ies found on the token
+	 * @deprecated Since 5.2. Use your own custom converter instead
+	 * @see JwtGrantedAuthoritiesConverter
+	 * @see #setGrantedAuthoritiesConverter(Converter)
+	 */
+	@Deprecated
 	protected Collection<GrantedAuthority> extractAuthorities(Jwt jwt) {
-		return this.getScopes(jwt)
-						.stream()
-						.map(authority -> SCOPE_AUTHORITY_PREFIX + authority)
-						.map(SimpleGrantedAuthority::new)
-						.collect(Collectors.toList());
+		return this.grantedAuthoritiesConverter.convert(jwt);
 	}
 
-	private Collection<String> getScopes(Jwt jwt) {
-		for ( String attributeName : WELL_KNOWN_SCOPE_ATTRIBUTE_NAMES ) {
-			Object scopes = jwt.getClaims().get(attributeName);
-			if (scopes instanceof String) {
-				if (StringUtils.hasText((String) scopes)) {
-					return Arrays.asList(((String) scopes).split(" "));
-				} else {
-					return Collections.emptyList();
-				}
-			} else if (scopes instanceof Collection) {
-				return (Collection<String>) scopes;
-			}
-		}
-
-		return Collections.emptyList();
+	/**
+	 * Sets the {@link Converter Converter&lt;Jwt, Collection&lt;GrantedAuthority&gt;&gt;} to use.
+	 * Defaults to {@link JwtGrantedAuthoritiesConverter}.
+	 *
+	 * @param grantedAuthoritiesConverter The converter
+	 * @since 5.2
+	 * @see JwtGrantedAuthoritiesConverter
+	 */
+	public void setGrantedAuthoritiesConverter(Converter<Jwt, Collection<GrantedAuthority>> grantedAuthoritiesConverter) {
```

Comment:
I apologize, but I missed this earlier. In `JwtAuthenticationProvider`, the setter is `setJwtAuthenticationConverter`. I think, then, it would be more consistent to call this `setJwtGrantedAuthoritiesConverter`.

Reply:
Will do @jzheaux. I'll also make the same change in `ReactiveJwtAuthenticationConverter` to be consistent.